// @expo/config-plugins/build/android/JsEngine.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.setJsEngine = exports.getJsEngine = exports.withJsEngineGradleProps = exports.DEFAULT_JS_ENGINE = exports.JS_ENGINE_PROP_KEY = void 0;
const android_plugins_1 = require("../plugins/android-plugins");
exports.JS_ENGINE_PROP_KEY = 'expo.jsEngine';
exports.DEFAULT_JS_ENGINE = 'jsc';
const withJsEngineGradleProps = config => {
    return android_plugins_1.withGradleProperties(config, config => {
        config.modResults = setJsEngine(config, config.modResults);
        return config;
    });
};
exports.withJsEngineGradleProps = withJsEngineGradleProps;
function getJsEngine(config) {
    var _a, _b;
    return (_b = (_a = config.android) === null || _a === void 0 ? void 0 : _a.jsEngine) !== null && _b !== void 0 ? _b : exports.DEFAULT_JS_ENGINE;
}
exports.getJsEngine = getJsEngine;
function setJsEngine(config, gradleProperties) {
    var _a, _b;
    const oldPropIndex = gradleProperties.findIndex(prop => prop.type === 'property' && prop.key === exports.JS_ENGINE_PROP_KEY);
    const newProp = {
        type: 'property',
        key: exports.JS_ENGINE_PROP_KEY,
        value: (_b = (_a = config.android) === null || _a === void 0 ? void 0 : _a.jsEngine) !== null && _b !== void 0 ? _b : exports.DEFAULT_JS_ENGINE,
    };
    if (oldPropIndex >= 0) {
        gradleProperties[oldPropIndex] = newProp;
    }
    else {
        gradleProperties.push(newProp);
    }
    return gradleProperties;
}
exports.setJsEngine = setJsEngine;
