// @expo/config-plugins/build/android/PrimaryColor.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.getPrimaryColor = exports.withPrimaryColorStyles = exports.withPrimaryColorColors = exports.withPrimaryColor = void 0;
const android_plugins_1 = require("../plugins/android-plugins");
const Colors_1 = require("./Colors");
const Styles_1 = require("./Styles");
const COLOR_PRIMARY_KEY = 'colorPrimary';
const DEFAULT_PRIMARY_COLOR = '#023c69';
const withPrimaryColor = config => {
    config = exports.withPrimaryColorColors(config);
    config = exports.withPrimaryColorStyles(config);
    return config;
};
exports.withPrimaryColor = withPrimaryColor;
const withPrimaryColorColors = config => {
    return android_plugins_1.withAndroidColors(config, config => {
        config.modResults = Colors_1.assignColorValue(config.modResults, {
            name: COLOR_PRIMARY_KEY,
            value: getPrimaryColor(config),
        });
        return config;
    });
};
exports.withPrimaryColorColors = withPrimaryColorColors;
const withPrimaryColorStyles = config => {
    return android_plugins_1.withAndroidStyles(config, config => {
        config.modResults = Styles_1.assignStylesValue(config.modResults, {
            add: !!getPrimaryColor(config),
            parent: Styles_1.getAppThemeLightNoActionBarGroup(),
            name: COLOR_PRIMARY_KEY,
            value: `@color/${COLOR_PRIMARY_KEY}`,
        });
        return config;
    });
};
exports.withPrimaryColorStyles = withPrimaryColorStyles;
function getPrimaryColor(config) {
    var _a;
    return (_a = config.primaryColor) !== null && _a !== void 0 ? _a : DEFAULT_PRIMARY_COLOR;
}
exports.getPrimaryColor = getPrimaryColor;
