// @expo/config-plugins/build/android/RootViewBackgroundColor.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.getRootViewBackgroundColor = exports.withRootViewBackgroundColorStyles = exports.withRootViewBackgroundColorColors = exports.withRootViewBackgroundColor = void 0;
const android_plugins_1 = require("../plugins/android-plugins");
const Colors_1 = require("./Colors");
const Styles_1 = require("./Styles");
const ANDROID_WINDOW_BACKGROUND = 'android:windowBackground';
const WINDOW_BACKGROUND_COLOR = 'activityBackground';
const withRootViewBackgroundColor = config => {
    config = exports.withRootViewBackgroundColorColors(config);
    config = exports.withRootViewBackgroundColorStyles(config);
    return config;
};
exports.withRootViewBackgroundColor = withRootViewBackgroundColor;
const withRootViewBackgroundColorColors = config => {
    return android_plugins_1.withAndroidColors(config, async (config) => {
        config.modResults = Colors_1.assignColorValue(config.modResults, {
            value: getRootViewBackgroundColor(config),
            name: WINDOW_BACKGROUND_COLOR,
        });
        return config;
    });
};
exports.withRootViewBackgroundColorColors = withRootViewBackgroundColorColors;
const withRootViewBackgroundColorStyles = config => {
    return android_plugins_1.withAndroidStyles(config, async (config) => {
        config.modResults = Styles_1.assignStylesValue(config.modResults, {
            add: !!getRootViewBackgroundColor(config),
            parent: Styles_1.getAppThemeLightNoActionBarGroup(),
            name: ANDROID_WINDOW_BACKGROUND,
            value: `@color/${WINDOW_BACKGROUND_COLOR}`,
        });
        return config;
    });
};
exports.withRootViewBackgroundColorStyles = withRootViewBackgroundColorStyles;
function getRootViewBackgroundColor(config) {
    var _a;
    return ((_a = config.android) === null || _a === void 0 ? void 0 : _a.backgroundColor) || config.backgroundColor || null;
}
exports.getRootViewBackgroundColor = getRootViewBackgroundColor;
